ntrans_dataprep.py

```diff
@@ -14,15 +14,23 @@
 
 """
 
+from __future__ import annotations
```

Comment:
After some research, I see that this is to prevent errors from type hinting returned variables that are not yet defined at the point of hinting. Do I understand this correctly, or are there more to say about this?

Reply:
Without this, `collections.Counter[str]` errors when running the code, but I want to use it to denote a counter of strings. So mypy understands `collections.Counter[str]`, but Python itself doesn't. It's basically a way to tell Python to ignore the type hints instead of trying to evaluate them as Python code, so if I use `collections.Counter[str]` in a type hint, Python doesn't care.

Reply:
Actually, I use `collections.Counter[NGram]`, but you get the idea.